### DISCUSSION TITLE
When running export on large database runs into a dead lock situation

### DISCUSSION THREAD

**Discussion by makpcheng** [JanusGraph/janusgraph#1323] [Q&A]
Version: Janusgraph 0.2.0

Steps:
Export graphml on a large database, one larger than 2GB in size.  Somewhere in between it will deadlock when writing out vertex objects and their properties. There appears to be a deadlock when querying for properties.

Workaround:
Add lots of debug statements to slow the export down enough to prevent the deadlock from happening.

**Comment by porunov** [comment]
It could be related to BerkleyDB issues but I didn't research it much. In case you used BerkleyDB as a storage backend, the next configuration may help in some situations:
``` 
storage.berkeleyje.lock-mode: LockMode.READ_UNCOMMITTED
```
That said, I also have seen many cases with lock acquittance timeout issues when a storage backend can't perform operations in a necessary time frame. I don't think this is directly connected to this problem, but the below configuration can help with timeouts:
```
storage.lock.wait-time: 5000
```
